src/main/java/uk/gov/justice/services/test/domain/EventsChecker.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.justice.services.test.domain;
+
+import static uk.gov.justice.services.test.DomainTest.eventNameFrom;
+import static uk.gov.justice.services.test.DomainTest.generatedEventAsJsonNode;
+import static uk.gov.justice.services.test.DomainTest.jsonNodeWithoutMetadataFrom;
+
+import uk.gov.justice.services.test.exception.EventException;
+
+import java.util.List;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.commons.lang3.StringUtils;
+
+public class EventsChecker {
+    public static void compareEvents(List<JsonNode> expectedEvents, List<Object> generatedEvents) {
+        if (expectedEvents.size() > generatedEvents.size()) {
+            throw new EventException(createMessage(expectedEvents, generatedEvents, null));
+        }
+
+        for (JsonNode jsonNode : expectedEvents) {
```

Comment:
The logic of this loop looks wrong. The expected events and actual events need to match exactly; we;re not searching for each expected event in the list of actual events, we want the two lists to be identical. I think you've changed the comparison being done and its not testing the behaviour we want any more.